### PR TITLE
Refactors species mutanthands into human component

### LINKED ISSUE #73034: Shattered ritual ghouls can regain use of both hands
Reporting client version: 514.1589

## Testmerges:
- [Don't initialize stack components inside machines](https://www.github.com/tgstation/tgstation/pull/72863)

## Reproduction:
As a shattered ritual ghoul, fireman carry a corpse. This will remove both of your big meaty claws and allow you to freely use both hands for the rest of the round.

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Mutanthands is now a component, which handles ensuring the mob always has their mutant hands equipped to their right slots at all times. 

Alternative to #73282 
Fixes #73034 

## Why It's Good For The Game

Cleaner and more consistent code, Ensures mutant hand mobs always have their correct hands. 

## Changelog

:cl: Melbert
fix: It should be way harder to lose your special hands as a zombie or shattered risen ghoul.
refactor: Refactored mutanthands for zombies and shattered risen.
/:cl:

### CODE REVIEW

**Comment by ChungusGamer666** [comment]
it's good that this is a component, but shouldn't mutant hands be handled by bodyparts instead of species?

**Comment by ChungusGamer666** [comment]
meh i guess i'll do yet another limb improvement pr once this is merged to avoid family guy funny moments

File: code/datums/components/mutant_hands.dm [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+		// This slot is not empty
+		// Yes the held item lists contains nulls to represent empty hands
+		// It saves us a /item cast by using as anything in the loop
+		if(!isnull(hand_slot))
+			if(HAS_TRAIT(hand_slot, TRAIT_NODROP) || (hand_slot.item_flags & ABSTRACT))
+				message_admins("[hand_slot] is nodrop or abstract")
+				// There's a nodrop / abstract item in the way of putting a mutant hand in
+				// It can stay, for now, but if it gets dropped / unequipped we'll swoop in to replace the slot
+				continue
+			message_admins("[hand_slot] can be dropped safely")
```
Comment (san7890) [inline code/datums/components/mutant_hands.dm:73]: Are these `message_admins` stuff for debugging, or do you actually want to pass this feedback along to admins? seems quite vague to be useful in production imo, would need to include more context to be useful.

File: code/datums/components/mutant_hands.dm [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+	if(QDELING(src) || QDELING(parent))
+		return
+
+	if(null in source.held_items)
+		INVOKE_ASYNC(src, PROC_REF(apply_mutant_hands))
+
+/**
+ * Generic mutant hand type for use with the mutant hands component
+ * (Technically speaking, the component doesn't require you use this type. B.ut it's here for posterity)
```
Comment (san7890) [inline code/datums/components/mutant_hands.dm:143]:
```suggestion
 * (Technically speaking, the component doesn't require you use this type. But it's here for posterity)
```

File: code/modules/mob/living/carbon/examine.dm [hunk 3]
```diff
@@ -19,8 +19,9 @@
 		. += "[t_He] [t_is] wearing [wear_neck.get_examine_string(user)] around [t_his] neck."
 
 	for(var/obj/item/I in held_items)
-		if(!(I.item_flags & ABSTRACT))
-			. += "[t_He] [t_is] holding [I.get_examine_string(user)] in [t_his] [get_held_index_name(get_held_index_of_item(I))]."
+		if(I.item_flags & (ABSTRACT|EXAMINE_SKIP|HAND_ITEM))
+			continue
+		. += "[t_He] [t_is] holding [I.get_examine_string(user)] in [t_his] [get_held_index_name(get_held_index_of_item(I))]."
```
Comment (san7890) [inline code/modules/mob/living/carbon/examine.dm:25]: clean up this single-letter var while in the area :)

File: code/modules/mob/living/carbon/human/examine.dm [hunk 4]
```diff
@@ -49,8 +49,9 @@
 
 	//Hands
 	for(var/obj/item/I in held_items)
-		if(!(I.item_flags & ABSTRACT) && !(I.item_flags & EXAMINE_SKIP))
-			. += "[t_He] [t_is] holding [I.get_examine_string(user)] in [t_his] [get_held_index_name(get_held_index_of_item(I))]."
+		if(I.item_flags & (ABSTRACT|EXAMINE_SKIP|HAND_ITEM))
+			continue
+		. += "[t_He] [t_is] holding [I.get_examine_string(user)] in [t_his] [get_held_index_name(get_held_index_of_item(I))]."
```
Comment (san7890) [inline code/modules/mob/living/carbon/human/examine.dm:55]: ditto single letter var

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm [hunk 5]
```diff
@@ -263,8 +263,9 @@
 
 	//Hands
 	for(var/obj/item/I in held_items)
-		if(!(I.item_flags & ABSTRACT))
-			. += "It has [I.get_examine_string(user)] in its [get_held_index_name(get_held_index_of_item(I))]."
+		if(I.item_flags & (ABSTRACT|EXAMINE_SKIP|HAND_ITEM))
+			continue
+		. += "It has [I.get_examine_string(user)] in its [get_held_index_name(get_held_index_of_item(I))]."
```
Comment (san7890) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm:268]: ditto single letter var

**Review by san7890 — CHANGES_REQUESTED** [review]

**Pull request merged**